### PR TITLE
INGK-670 Implement EtherCAT SDO register and dictionary classes

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Implement EtherCAT SDO register and dictionary classes.

Fixes #INGK-670

### Type of change

- Add EthercatDictionary class.
- Add EthercatRegister class.

### Tests
- Load an Ethercat dictionary (e_eoe) and check that the MCB addresses are mapped correctly to the CANopen addresses.

### CODE REVIEW

File: ingenialink/ethercat/dictionary.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from ingenialink.dictionary import Dictionary
+from ingenialink.ethercat.register import EthercatRegister
+from ingenialink.constants import (
+    CANOPEN_ADDRESS_OFFSET,
+    CANOPEN_SUBNODE_0_ADDRESS_OFFSET,
+    MAP_ADDRESS_OFFSET,
+)
+
+import ingenialogger
+import xml.etree.ElementTree as ET
```
Comment (WoodElv) [inline ingenialink/ethercat/dictionary.py:12]: Order import is not correct.

File: ingenialink/ethercat/register.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        address_type: Address Type.
+
+    Raises:
+        TypeError: If any of the parameters has invalid type.
+        ILValueError: If the register is invalid.
+        ILAccessError: Register with wrong access type.
+
+    """
+
+    def __init__(
```
Comment (WoodElv) [inline ingenialink/ethercat/register.py]: You can delete the whole constructor and just put a `pass` instead.
In addition, as the constructor is the same, you can delete the `Args` and `Raises` fields in docstrings.

**Review by WoodElv — CHANGES_REQUESTED** [review]

File: ingenialink/ethercat/register.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/ethercat/register.py]: Done

File: ingenialink/ethercat/dictionary.py (hunk 1 above)
Comment (MartinALL91) [reply ingenialink/ethercat/dictionary.py]: Done

**Review by WoodElv — APPROVED** [review]

**Review by pzinemanas — APPROVED** [review]

**Pull request merged**